### PR TITLE
[Validator] Fix the locale validator so it treats a locale alias as a valid locale

### PR DESCRIPTION
**Pull request by jakzal**
| Q | A |
| --- | --- |
| Branch | 2.3 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #12583 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Validator/composer.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "require-dev": {
         "doctrine/common": "~2.3",
         "symfony/http-foundation": "~2.1",
-        "symfony/intl": "~2.3",
+        "symfony/intl": "^2.3.21",
```
Comment (jakzal) [inline src/Symfony/Component/Validator/composer.json]: `LocaleBundle::getAliases()` was introduced in Symfony 2.3.21.

File: src/Symfony/Component/Validator/Constraints/LocaleValidator.php [hunk 2]
```diff
@@ -38,8 +38,9 @@ public function validate($value, Constraint $constraint)
 
         $value = (string) $value;
         $locales = Intl::getLocaleBundle()->getLocaleNames();
+        $aliases = Intl::getLocaleBundle()->getAliases();
 
-        if (!isset($locales[$value])) {
+        if (!isset($locales[$value]) && !array_search($value, $aliases)) {
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/Constraints/LocaleValidator.php]: As we are not interested in the index wouldn't it be make sense to use `in_array()` instead?

Comment (jakzal) [inline src/Symfony/Component/Validator/Constraints/LocaleValidator.php]: Obviously :)

**Comment by xabbuh** [comment]
:+1: 

Status: Reviewed

**Comment by webmozart** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @jakzal.

**Pull request merged**